src/test/ui/question-mark-type-infer.stderr

```diff
@@ -1,8 +1,13 @@
-error[E0284]: type annotations needed: cannot resolve `<_ as std::ops::Try>::Ok == _`
-  --> $DIR/question-mark-type-infer.rs:12:5
+error[E0284]: type annotations needed
+  --> $DIR/question-mark-type-infer.rs:12:21
    |
 LL |     l.iter().map(f).collect()?
-   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |                     ^^^^^^^
+   |                     |
+   |                     cannot infer type
+   |                     help: consider specifying the type argument in the method call: `collect::<_>`
```

Comment:
do we know that specifying the type argument will help? I have to go check back in the code, I guess

Reply:
No, we don't it's a stab in the dark and I haven't seen a way to get the `FnDecl` corresponding to `collect` (I haven't been able to procure a `DefId`, even). I'm ok with not giving this suggestion in this PR and landing the rest, but this was the entire reason I looked at this in the first place.

Reply:
^ re: my other recent comment